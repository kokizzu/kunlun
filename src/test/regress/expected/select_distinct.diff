60,61c60,61
<                                                          QUERY PLAN                                                         
< ----------------------------------------------------------------------------------------------------------------------------
---
>                        QUERY PLAN                       
> --------------------------------------------------------
64c64
<    ->  Unique
---
>    ->  HashAggregate
66c66,67
<          ->  RemoteScan on public.tenk1
---
>          Group Key: tenk1.two, tenk1.four, tenk1.two
>          ->  Seq Scan on public.tenk1
68d68
<                Shard: 2  Remote SQL: select  tenk1.two, tenk1.four from regress_$$_public.tenk1 tenk1 ORDER BY 1 ASC, 2 ASC
