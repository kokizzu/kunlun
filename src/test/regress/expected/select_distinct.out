--
-- SELECT_DISTINCT
--
COPY tenk1 FROM '/home/dzw/work/kunlun-enterprise/postgresql-11.5/src/test/regress/data/tenk.data';
psql:sql/select_distinct.sql:5: ERROR:  could not open file "/home/dzw/work/kunlun-enterprise/postgresql-11.5/src/test/regress/data/tenk.data" for reading: No such file or directory
HINT:  COPY FROM instructs the PostgreSQL server process to read a file. You may want a client-side facility such as psql's \copy.
--
-- awk '{print $3;}' onek.data | sort -n | uniq
--
SELECT DISTINCT two FROM tenk1 ORDER BY 1;
 two 
-----
(0 rows)

--
-- awk '{print $5;}' onek.data | sort -n | uniq
--
SELECT DISTINCT ten FROM tenk1 ORDER BY 1;
 ten 
-----
(0 rows)

--
-- awk '{print $16;}' onek.data | sort -d | uniq
--
SELECT DISTINCT string4 FROM tenk1 ORDER BY 1;
 string4 
---------
(0 rows)

--
-- awk '{print $3,$16,$5;}' onek.data | sort -d | uniq |
-- sort +0n -1 +1d -2 +2n -3
--
SELECT DISTINCT two, string4, ten
   FROM tenk1
   ORDER BY two using <, string4 using <, ten using <;
 two | string4 | ten 
-----+---------+-----
(0 rows)

--
-- awk '{print $2;}' person.data |
-- awk '{if(NF!=1){print $2;}else{print;}}' - emp.data |
-- awk '{if(NF!=1){print $2;}else{print;}}' - student.data |
-- awk 'BEGIN{FS="      ";}{if(NF!=1){print $5;}else{print;}}' - stud_emp.data |
-- sort -n -r | uniq
--
SELECT DISTINCT p.age FROM person* p ORDER BY age using >;
 age 
-----
(0 rows)

--
-- Check mentioning same column more than once
--
EXPLAIN (VERBOSE, COSTS OFF)
SELECT count(*) FROM
  (SELECT DISTINCT two, four, two FROM tenk1) ss;

SELECT count(*) FROM
  (SELECT DISTINCT two, four, two FROM tenk1) ss;tenk1
 count 
-------
     0
(1 row)

--
-- Also, some tests of IS DISTINCT FROM, which doesn't quite deserve its
-- very own regression file.
--
CREATE TEMP TABLE disttable (f1 integer);
psql:sql/select_distinct.sql:54: ERROR:  syntax error at or near "tenk1"
LINE 1: tenk1
        ^
INSERT INTO DISTTABLE VALUES(1);
psql:sql/select_distinct.sql:55: ERROR:  relation "disttable" does not exist
LINE 1: INSERT INTO DISTTABLE VALUES(1);
                    ^
INSERT INTO DISTTABLE VALUES(2);
psql:sql/select_distinct.sql:56: ERROR:  relation "disttable" does not exist
LINE 1: INSERT INTO DISTTABLE VALUES(2);
                    ^
INSERT INTO DISTTABLE VALUES(3);
psql:sql/select_distinct.sql:57: ERROR:  relation "disttable" does not exist
LINE 1: INSERT INTO DISTTABLE VALUES(3);
                    ^
INSERT INTO DISTTABLE VALUES(NULL);
psql:sql/select_distinct.sql:58: ERROR:  relation "disttable" does not exist
LINE 1: INSERT INTO DISTTABLE VALUES(NULL);
                    ^
-- basic cases
SELECT f1, f1 IS DISTINCT FROM 2 as "not 2" FROM disttable;
psql:sql/select_distinct.sql:61: ERROR:  relation "disttable" does not exist
LINE 1: SELECT f1, f1 IS DISTINCT FROM 2 as "not 2" FROM disttable;
                                                         ^
SELECT f1, f1 IS DISTINCT FROM NULL as "not null" FROM disttable;
psql:sql/select_distinct.sql:62: ERROR:  relation "disttable" does not exist
LINE 1: ...T f1, f1 IS DISTINCT FROM NULL as "not null" FROM disttable;
                                                             ^
SELECT f1, f1 IS DISTINCT FROM f1 as "false" FROM disttable;
psql:sql/select_distinct.sql:63: ERROR:  relation "disttable" does not exist
LINE 1: SELECT f1, f1 IS DISTINCT FROM f1 as "false" FROM disttable;
                                                          ^
SELECT f1, f1 IS DISTINCT FROM f1+1 as "not null" FROM disttable;
psql:sql/select_distinct.sql:64: ERROR:  relation "disttable" does not exist
LINE 1: ...T f1, f1 IS DISTINCT FROM f1+1 as "not null" FROM disttable;
                                                             ^
-- check that optimizer constant-folds it properly
SELECT 1 IS DISTINCT FROM 2 as "yes";
 yes 
-----
 t
(1 row)

SELECT 2 IS DISTINCT FROM 2 as "no";
 no 
----
 f
(1 row)

SELECT 2 IS DISTINCT FROM null as "yes";
 yes 
-----
 t
(1 row)

SELECT null IS DISTINCT FROM null as "no";
 no 
----
 f
(1 row)

-- negated form
SELECT 1 IS NOT DISTINCT FROM 2 as "no";
 no 
----
 f
(1 row)

SELECT 2 IS NOT DISTINCT FROM 2 as "yes";
 yes 
-----
 t
(1 row)

SELECT 2 IS NOT DISTINCT FROM null as "no";
 no 
----
 f
(1 row)

SELECT null IS NOT DISTINCT FROM null as "yes";
 yes 
-----
 t
(1 row)

